packages/storybook/src/executors/storybook/storybook.impl.ts

```diff
@@ -26,6 +27,21 @@ export default async function* storybookExecutor(
 ) {
   let frameworkPath = getStorybookFrameworkPath(options.uiFramework);
 
+  if (
+    !options.useDefaultProject &&
+    context.projectName &&
+    options.uiFramework === '@storybook/angular'
+  ) {
+    const buildable =
+      !!context?.workspace?.projects?.[context.projectName]?.targets?.build;
+    options = {
+      ...options,
+      projectBuildConfig: `${context.projectName}${
```

Comment:
I think these changes should go somewhere in here: https://github.com/nrwl/nx/blob/eb9a7135e2b91bbef9032ae5fd5f05ee5aa5cb07/packages/storybook/src/executors/utils.ts#L188-L232

There's already a logic for resolving the default project and communicating it to the Storybook internal builder.